diff --git a/include/menuitems_numbers.h b/include/menuitems_numbers.h
index 2ac25d4..e54fafb 100644
--- a/include/menuitems_numbers.h
+++ b/include/menuitems_numbers.h
@@ -101,20 +101,20 @@ class NumberControl : public NumberControlBase {
                 sprintf(fmt, "%-5i", value);
             return fmt;
         }
-        virtual const char *getFormattedValue(uint32_t value) {
+        virtual const char *getFormattedValue(unsigned int value) {
             static char fmt[MENU_C_MAX] = "      ";
             if (this->debug)
-                sprintf(fmt, "%-5u [ulong]", value);
+                sprintf(fmt, "%-5u [ulong]", (unsigned int) value);
             else
-                sprintf(fmt, "%-5u", value);
+                sprintf(fmt, "%-5u", (unsigned int) value);
             return fmt;
         }
-        virtual const char *getFormattedValue(int32_t value) {
+        virtual const char *getFormattedValue(signed long value) {
             static char fmt[MENU_C_MAX] = "      ";
             if (this->debug)
-                sprintf(fmt, "%-5i [long]", value);
+                sprintf(fmt, "%-5i [int]", (int) value);
             else
-                sprintf(fmt, "%-5i", value);
+                sprintf(fmt, "%-5i", (int) value);
             return fmt;
         }
         virtual const char *getFormattedValue(double value) {
diff --git a/include/menuitems_object.h b/include/menuitems_object.h
index 73f0d32..05c0607 100644
--- a/include/menuitems_object.h
+++ b/include/menuitems_object.h
@@ -90,7 +90,7 @@ class ObjectNumberControl : public NumberControl<DataType> {
 
             char msg[255];
             //Serial.printf("about to build msg string...\n");
-            sprintf(msg, "Set %8s to %i", this->label, value);
+            sprintf(msg, "Set %8s to %i", this->label, (int)value);
             //Serial.printf("about to set_last_message!");
             msg[this->tft->get_c_max()] = '\0'; // limit the string so we don't overflow set_last_message
             menu_set_last_message(msg,GREEN);
diff --git a/include/menuitems_object_selector.h b/include/menuitems_object_selector.h
index ffb981e..cac5e7e 100644
--- a/include/menuitems_object_selector.h
+++ b/include/menuitems_object_selector.h
@@ -109,7 +109,7 @@ class ObjectSelectorControl : public ObjectNumberControl<TargetClass,DataType> {
 
             char msg[255];
             //Serial.printf("about to build msg string...\n");
-            sprintf(msg, "Set %8s to %i", this->label, value);
+            sprintf(msg, "Set %8s to %i", this->label, (int)value);
             //Serial.printf("about to set_last_message!");
             msg[this->tft->get_c_max()] = '\0'; // limit the string so we don't overflow set_last_message
             menu_set_last_message(msg,GREEN);
